qiskit/utils/deprecation.py

```diff
@@ -246,8 +248,12 @@ def _write_deprecation_msg(
     since: str,
     pending: bool,
     additional_msg: str,
-    removal_timeline: str,
+    removal_timeline: str | None,
 ) -> tuple[str, Type[DeprecationWarning] | Type[PendingDeprecationWarning]]:
+    if not removal_timeline:
+        removal_major = int(since.split(".")[0]) + 1
```

Comment:
I'm a bit concerned this might break people since we had two instances of using a float in Qiskit. Could you change this to:

```suggestion
        removal_major = int(str(since).split(".")[0]) + 1
```

just as a a little protection. The API clearly documents that `since` should be a string, but this doesn't seem like a terrible way to prevent accidental breakage.